src/components/Button/Button.scss

```diff
@@ -80,32 +80,43 @@ $stacking-order: (
     border-top-right-radius: 0;
     border-bottom-right-radius: 0;
 
+    &::after {
+      border-top-right-radius: 0;
+      border-bottom-right-radius: 0;
+    }
+
     &:focus {
       z-index: z-index(focused, $stacking-order);
     }
   }
 }
-// stylelint-disable selector-max-combinators
+
+// stylelint-disable selector-max-combinators, selector-max-compound-selectors, selector-max-specificity
 [data-buttongroup-segmented] {
-  .Button {
+  .Button,
+  .Button::after {
     border-radius: 0;
   }
-  > :first-child .Button {
+  > :first-child .Button,
+  > :first-child .Button::after {
```

Comment:
Easy peasy. I didn't bother adding the .globalTheming here because this doesn't impact the non-themable buttons.